spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -136,10 +150,13 @@ protected Object doReceive() {
 			String remotePath = remotePath(file);
 			Session<?> session = this.remoteFileTemplate.getSession();
 			try {
-				return getMessageBuilderFactory().withPayload(session.readRaw(remotePath))
+				return getMessageBuilderFactory()
+						.withPayload(session.readRaw(remotePath))
 						.setHeader(IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE, session)
 						.setHeader(FileHeaders.REMOTE_DIRECTORY, file.getRemoteDirectory())
 						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename())
+						.setHeader(FileHeaders.REMOTE_FILE_INFO,
+										this.fileInfoJson ? (file.toJson()) : file.getFileInfo())
```

Comment:
Why `file.getFileInfo()`?
Isn't consistent with `file.toJson()`.
So, we are expecting the `FileInfo` object there, not low-level entity.